v2/dtos/requests/deviceservice_test.go

```diff
@@ -115,3 +138,122 @@ func Test_AddDeviceServiceReqToDeviceServiceModels(t *testing.T) {
+	}
+	fmt.Println(string(resultTestBytes))
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var expected = tt.addDeviceService
+			err := tt.addDeviceService.UnmarshalJSON(tt.args.data)
+			if tt.wantErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, expected, tt.addDeviceService, "Unmarshal did not result in expected AddDeviceServiceRequest.", err)
+			}
+		})
+	}
+}
+
+func TestUpdateDeviceServiceRequest_Validate(t *testing.T) {
+	valid := testUpdateDeviceService
+	validWithoutId := valid
+	validWithoutId.Service.Id = nil
+	validWithoutName := valid
+	validWithoutName.Service.Name = nil
+	noReID := valid
+	noReID.RequestID = ""
+	noIdAndName := valid
+	noIdAndName.Service.Id = nil
+	noIdAndName.Service.Name = nil
+	invalidOperatingState := valid
+	invalid := "invalid"
+	invalidOperatingState.Service.OperatingState = &invalid
+	invalidAdminState := valid
+	invalidAdminState.Service.OperatingState = &invalid
+	tests := []struct {
+		name          string
+		DeviceService UpdateDeviceServiceRequest
+		expectError   bool
+	}{
+		{"valid AddDeviceServiceRequest", valid, false},
+		{"valid AddDeviceServiceRequest without Id", validWithoutId, false},
+		{"valid AddDeviceServiceRequest without name", validWithoutName, false},
+		{"invalid AddDeviceServiceRequest, no Request Id", noReID, true},
+		{"invalid AddDeviceServiceRequest, no Id and Name", noIdAndName, true},
+		{"invalid AddDeviceServiceRequest, invalid OperatingState", invalidOperatingState, true},
+		{"invalid AddDeviceServiceRequest, invalid AdminState", invalidAdminState, true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.DeviceService.Validate()
+			assert.Equal(t, tt.expectError, err != nil, "Unexpected updateDeviceServiceRequest validation result.", err)
+		})
+	}
+}
+
+func TestUpdateDeviceServiceRequest_UnmarshalJSON_NilField(t *testing.T) {
+	reqJson := `{
+        "requestId":"7a1707f0-166f-4c4b-bc9d-1d54c74e0137",
+		"service":{
+			"name":"test device service"
+		}
+	}`
+	var req UpdateDeviceServiceRequest
+
+	err := req.UnmarshalJSON([]byte(reqJson))
+
+	assert.Equal(t, nil, err)
+	// Nil field checking is used to update with patch
+	assert.True(t, req.Service.BaseAddress == nil)
+	assert.True(t, req.Service.AdminState == nil)
+	assert.True(t, req.Service.OperatingState == nil)
+	assert.True(t, req.Service.Labels == nil)
+}
+
+func TestUpdateDeviceServiceRequest_UnmarshalJSON_EmptySlice(t *testing.T) {
+	reqJson := `{
+        "requestId":"7a1707f0-166f-4c4b-bc9d-1d54c74e0137",
+		"service":{
+			"name":"test device",
+			"labels":[]
+		}
+	}`
+	var req UpdateDeviceServiceRequest
+
+	err := req.UnmarshalJSON([]byte(reqJson))
+
+	assert.Equal(t, nil, err)
+	// Empty slice is used to remove the data
+	assert.True(t, req.Service.Labels != nil)
```

Comment:
use `assert.NotNil(...)`

Reply:
Finished all request changes. 
Thanks for catching lots of typo and misused.